code/data_structures/linked_list/linked_lists_operations/linked_list_cycle.cpp

```diff
@@ -0,0 +1,64 @@
+/* This code detects and removes cycle in a linked list,
+	it also transforms circular lists into linear lists.
+
+	Data Entry: Beginning
+	Cycle Placement: Nth Position
+	Cycle Detection: Nth Position
+	Cycle Removal: Nth Position
+
+	By: Sarthi Chugh (user: sarthi92)
+		BITS, Pilani
+*/
+#ifndef _IOSTREAM_H
+#include<iostream>
+#endif
+using namespace std;
+struct Node
+{
+    int data;
+    Node* next;
+}*head;
+
+int linked_list_operations(int ch,int x)
+{
+    Node *temp1,*temp2;
+    Node *n;
+    switch(ch)
+    {
+    case 1://Node Insertion with data=x at beginning of the List
+        n=new Node;
+        n->data=x;
+        n->next=head;
+        head=n;
+        break;
+    case 2://Cycle Placement at position=x
+        if(head!=NULL)
+        {
+        for(temp1=head;temp1->next!=NULL;temp1=temp1->next);
+        for(temp2=head;x>1;x--,temp2=temp2->next);
+        temp1->next=temp2;

```

Comment:
Fix indentation.

Reply:
indentation not required here as for statement terminated  with ;

Reply:
Oh right, missed that.